docs/iris/src/further_topics/lenient_metadata.rst

```diff
@@ -0,0 +1,390 @@
+    {'grinning face': '😀', 'neutral face': '😐'}
+
+We create another :class:`~iris.common.DimCoordMetadata` with a dissimilar
+``attributes`` member, namely,
+
+- the ``grinning face`` key is **missing**,
+- the ``neutral face`` key has a **different value**, and
+- the ``upside-down face`` key is **new**
+
+.. doctest:: lenient-behaviour
+
+    >>> attributes = {"neutral face": "😜", "upside-down face": "🙃"}
+    >>> metadata = latitude.metadata._replace(attributes=attributes)
+    >>> metadata  # doctest: +SKIP
+    DimCoordMetadata(standard_name='latitude', long_name=None, var_name='latitude', units=Unit('degrees'), attributes={'neutral face': '😜', 'upside-down face': '🙃'}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+Now comparing the strict and lenient behaviour for the ``difference`` method,
+highlights the change in how such dissimilar metadata is treated gracefully,
+
+.. doctest:: lenient-behaviour
+
+    >>> metadata.difference(latitude.metadata).attributes  # doctest: +SKIP
+    {'upside-down face': '🙃', 'neutral face': '😜'}, {'neutral face': '😐', 'grinning face': '😀'}
+    >>> metadata.difference(latitude.metadata, lenient=True).attributes  # doctest: +SKIP
+    {'neutral face': '😜'}, {'neutral face': '😐'}
+
+
+.. _lenient combination:
+
+Lenient combination
+-------------------
+
+The behaviour of the lenient ``combine`` metadata class method is outlined
+in :numref:`lenient combine table`, and as with :ref:`lenient equality` and
+:ref:`lenient difference` is enabled throught the ``lenient`` keyword argument.
+
+The difference in behaviour between **lenient** and
+:ref:`strict combination <strict combine table>` is centered around the lenient
+handling of combining **something** with **nothing** (``None``) to return
+**something**. Whereas strict
+combination will only return a result from combining identical objects.
+
+Again, this is best demonstrated through a simple example of attempting to combine
+partially overlapping ``attributes`` member dictionaries. For example, given the
+following ``attributes`` dictionary of our favoured ``latitude`` coordinate,
+
+.. doctest:: lenient-behaviour
+
+    >>> latitude.attributes  # doctest: +SKIP
+    {'grinning face': '😀', 'neutral face': '😐'}
+
+We create another :class:`~iris.common.DimCoordMetadata` with overlapping
+keys and values, namely,
+
+- the ``grinning face`` key is **missing**,
+- the ``neutral face`` key has the **same value**, and
+- the ``upside-down face`` key is **new**
+
+.. doctest:: lenient-behaviour
+
+    >>> attributes = {"neutral face": "😐", "upside-down face": "🙃"}
+    >>> metadata = latitude.metadata._replace(attributes=attributes)
+    >>> metadata  # doctest: +SKIP
+    DimCoordMetadata(standard_name='latitude', long_name=None, var_name='latitude', units=Unit('degrees'), attributes={'neutral face': '😜', 'upside-down face': '🙃'}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
```

Comment:
I don't understand where 😜 is coming from in this example? Should this be 😐 since that is what's in both `attributes` and `latitude.attributes`?

Reply:
@stephenworsley Don't worry, my fault and awesome spot, thanks!

I was flipping on that `doctest` and forgot to update the output, so that's why it's confusing, simply because it's wrong. Since I'm using dictionaries I need to turn `doctest` off to account for dictionaries not being ordered, otherwise there is the risk of the doc-tests failing at some random point.

Pushed an update now, so that should have resolved this :+1: